### ISSUE TITLE
vue-runtime-helpers/dist/normalize-component.js not found - 2.17.0

### ISSUE THREAD

**Issue by chasegiunta** [xaksis/vue-good-table#569]
### Issue Type
- [x] Bug

### Specs
What version are you using?
2.17.0

Technically 2.17.0 doesn't seem to be released yet, but dependabot flagged the updated dependency for me, so it might've for others as well - so I'll just let you know anyways.

After updating, build/dev server fails on 2.17.0

```
This dependency was not found:

* vue-runtime-helpers/dist/normalize-component.js in ./node_modules/vue-good-table/dist/vue-good-table.esm.js

To install it, you can run: npm install --save vue-runtime-helpers/dist/normalize-component.js
```
I couldn't manually install `vue-runtime-helpers/dist/normalize-component.js` either - returned 405 error...

Downgrading to 2.16.5 fixes for now!

**Comment by xaksis** [comment]
looks like an error in rollup plugin
https://github.com/vuejs/rollup-plugin-vue/issues/257
I updated the dependency. Thank you for bringing this up.

**Comment by stefanwuthrich** [comment]
@xaksis I had the same problem, after upgrade to 2.7.1 I get the same error as before "fix".

**Comment by stefanwuthrich** [comment]
@chasegiunta can you confirm that it is fixed for you ih 2.7.1 ? (as for me it is not)

**Comment by ekrist1** [comment]
I got the same problem with 2.7.1 running npm run dev (Laravel + Vue installation):
Module not found: Error: Can't resolve 'vue-runtime-helpers/dist/normalize-component.js' in '/Users/myuser/Code/hvaskjer/node_modules/vue-good-table/dist'

**Comment by xaksis** [comment]
Going to reopen this to investigate

**Comment by xaksis** [comment]
followed the fix suggested here: 
https://stackoverflow.com/questions/54379726/rollup-plugin-vue-v4-6-2-has-a-problem-with-vue-runtime-helpers/54435814#54435814

**Comment by rakk7** [comment]
still having the same issue for version `2.17.2`

**Comment by ccclin** [comment]
One of workaround is add `devDependencies` into package.json
like:
```
"devDependencies": {
    ...
    "rollup-plugin-commonjs": "^10.0.1",
    "rollup-plugin-vue": "^5.0.1",
    ...
}
```

**Comment by rashed50702** [comment]
> One of workaround is add `devDependencies` into package.json
> like:
> 
> ```
> "devDependencies": {
>     ...
>     "rollup-plugin-commonjs": "^10.0.1",
>     "rollup-plugin-vue": "^5.0.1",
>     ...
> }
> ```

Still it's not working !!

**Comment by ccclin** [comment]
> > One of workaround is add `devDependencies` into package.json
> > like:
> > ```
> > "devDependencies": {
> >     ...
> >     "rollup-plugin-commonjs": "^10.0.1",
> >     "rollup-plugin-vue": "^5.0.1",
> >     ...
> > }
> > ```
> 
> Still it's not working !!

It works for my project and I am not sure if need to install additional dependencies.

**Comment by xaksis** [comment]
darn! these dependency issues will be the end of me... specially because i'm having trouble reproducing this on my end. re-opening.

**Comment by xaksis** [comment]
same as #572

**Comment by xaksis** [comment]
Let's see what egoist suggests: https://github.com/egoist/bili/issues/221

**Comment by controljud** [comment]
> One of workaround is add `devDependencies` into package.json
> like:
> 
> ```
> "devDependencies": {
>     ...
>     "rollup-plugin-commonjs": "^10.0.1",
>     "rollup-plugin-vue": "^5.0.1",
>     ...
> }
> ```

Same problem, but this works for me
Thanks

**Comment by ashniazi** [comment]
> "rollup-plugin-commonjs": "^10.0.1",
>     "rollup-plugin-vue": "^5.0.1",

Can confirm this fixed my problem too.

**Comment by sheldoncoupeheure** [comment]
yarn user here
following @ccclin's solution, deleting `node_modules` and operating a fresh `yarn install` solved the problem.
thanks

**Comment by gabrieldarezzo** [comment]
Same issue here: 
```
This dependency was not found:

* vue-runtime-helpers/dist/normalize-component.js in ./node_modules/vue-good-table/dist/vue-good-table.esm.js

To install it, you can run: npm install --save vue-runtime-helpers/dist/normalize-component.js
````  
This just happen after run:
`npm update`

I checked my `package-lock.json`
`"vue-good-table": "2.17.2"`

I just rollback to:
```
...
"dependencies": {
  "vue-good-table": "2.16.5",
}
...
```
Run:
`npm install`

And everthing works again ^^

**Comment by mesqueeb** [comment]
Same problem here.
Not using Rollup, only webpack.
So the current solution is to use 2.16.5?

**Comment by xaksis** [comment]
Yes, please continue to use 2.16.5 for now. I'll update once I have a good solution for the rollup changes.

**Comment by jorodriguez** [comment]
Solved with,

"rollup-plugin-commonjs": "^ 10.0.1", 
"rollup-plugin-vue": "^ 5.0.1",

and npm install.

tank's

**Comment by comhernandez28** [comment]
I had this exact problem in in an Installation of Laravel 5.8.28/Vue 2.6.10
I did not have rollup installed so I'm not sure of what the cause is but I just ended up changing the import to this
`import VueGoodTable from 'vue-good-table/dist/vue-good-table.js'; `
Which ended up working. 
Not sure if this is a proper fix though.

**Comment by xaksis** [comment]
@comhernandez28 yeah, the issue is only with the esm file that is created. umd and cjs versions should work fine.

**Comment by xaksis** [comment]
inlining the missing dependency in 2.17.3. This should fix the not found issue.

**Comment by chasegiunta** [comment]
@xaksis working on my end 👍- but I'll leave to you to close if you want to hear from others.

**Comment by mesqueeb** [comment]
Does "inlining" mean the library grew in size?

**Comment by zcuric** [comment]
@xaksis Hi, I have the same issue on my component for Plyr (https://github.com/zcuric/plyrue/issues/5).

Your solution to bundle node modules with `bili` ended up increasing the size of bundle for `vue-good-table`. Comparison can be made here on  [unpkg](https://unpkg.com/browse/vue-good-table@2.18.0/dist/) or on [bundlephobia](https://bundlephobia.com/result?p=vue-good-table@2.18.0). 

I suggest doing what I did, until proper solution has been found, lock `rollup-plugin-vue` to `4.3.2`. You'll keep your build smaller. 

I'll try to fix this in a better way and inform you if I found anything. Cheers.

**Comment by LeCoupa** [comment]
@zcuric did you find a better way?

**Comment by zcuric** [comment]
@LeCoupa To be honest, I didn't look for it, didn't have time. I'll will take a look in the nearby future.

**Comment by LeCoupa** [comment]
@zcuric I ended up using `vue-sfc-rollup`. It made the process of bundling components way simpler.

https://github.com/team-innovation/vue-sfc-rollup
